Doc for setting up a new environment name

Yivan opened the pull request:
This PR is to give clues how to configure a new environment name.

Even if it follow symfony standard, i think it is worth mentionning this in the documentation to help people (and those comming from Pimcore 4 can get trapped in the test environment which has a special signification in Symfony).

I state too that some environment file need to be created in `pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/` which seems is not documented.

This PR was done because of those issues relating some difficulties to set a new environment:
https://github.com/pimcore/pimcore/issues/2265
https://github.com/pimcore/pimcore/issues/2346

kkarkus commented on doc/Development_Documentation/21_Deployment/03_Multi_Environment.md:
````diff
@@ -33,16 +33,62 @@ var/config/system.php
+
+```
+app/config/pimcore/
+pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/
````
@Yivan does it mean I have to create config inside core dir?

mgomse commented on doc/Development_Documentation/21_Deployment/03_Multi_Environment.md:
````diff
@@ -33,16 +33,62 @@ var/config/system.php
+
+```
+app/config/pimcore/
+pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/
````
Changes in the core isn't good idea IMHO. Isn't it possible to do it in some other way?

dpfaffenbauer replied:
Shouldn't be necessary at all, ```app/config/pimcore``` or ```src/AppBundle/Resources/config``` should be totally enough

maff replied:
@Yivan the only necessary file is `app/config/config_<environment>.yml`. Please do not suggest adding/changing core files. IMHO the previous description was fine - I'd just add the note regarding the test environment.

Yivan commented:
I totally agree, it is strange to have to change core files, but it was needed when i tried (see the monlog error here https://github.com/pimcore/pimcore/issues/2265 please).
I will try again with the last version to see if this problem always occurs.

Maybe there is some special configuration flags to add in the new Yaml file created for the new environment (`app/config/config_<environment>.yml`) so no more monolog error is triggered ?
Just copying `app/config/config_dev.yml` to `app/config/config_mynewenv.yml` don't work.

Maybe a file config_env.yml.sample could be added as a starting point ?

Yivan commented:
I updated the documentation, so there is no more need to create yaml file in the core bundle.
Instead i added some exemple/hints, so developper can create new environment like dev or prod as starting base.

If all is ok, this can be merged, thanks.

maff commented on doc/Development_Documentation/21_Deployment/03_Multi_Environment.md:
```diff
@@ -33,16 +33,126 @@ var/config/system.php
+				$collection->addBundle(
+					new SensioGeneratorBundle(),
+					80, // priority
+					['dev'] // will use the dev config files of the bundle
```
This means the bundle will only be loaded in the `dev` environment and should be changed to `staging` in your example (comment should be removed or changed).

maff commented:
@Yivan I restructured the core config files in #2400, so they are now easily includable from a custom env config. This avoids having to replicate all the config for a specific environment (see the comment on the PR). Could you update this PR accordingly?

maff commented:
@Yivan any update on this?

Yivan commented:
@maff I will soon update my PR before the end of the week, is it ok ? Thanks.

Yivan commented:
@maff Code has just been updated, if all is ok you can merge it! Thanks.

maff commented on doc/Development_Documentation/21_Deployment/03_Multi_Environment.md:
```diff
@@ -33,16 +33,64 @@ var/config/system.php
+				$collection->addBundle(
+					new SensioGeneratorBundle(),
+					80, // priority
+					['staging'] // will use the staging config files of the bundle
```
@Yivan could you change this to something like `the bundle will only be loaded in the staging environment`? It does not actually change how config files are handled, just sets a filter when loading the bundle.

Yivan commented:
@maff change on the comment has been done, all should be ok now.

maff commented:
Thanks!
